File: 백준/Silver/16953. A → B/A → B.py
A, B = map(int, input().split())

result = 0
while True:
  if A == B:
    result += 1
    break
  elif A > B:
    result = -1
    break
  else:
    if B % 2 == 0:
      # B를 2로 나누기
      B //= 2
      result += 1
    else:
      # 일의 자리 1인 경우
      if str(B)[-1] == '1':
        b = str(B)
        B = int(b[:len(b)-1])
        result += 1
      else:
        result = -1
        break

print(result)

File: 백준/Silver/16953. A → B/A → B.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let [A, B] = input[0].split(" ").map(Number);
let result = 0;

while (A != B) {
  if (A > B) {
    result = -2;
    break;
  }
  if (B % 2 == 0) B /= 2;
  else {
    let str_B = String(B);
    if (str_B[str_B.length - 1] == "1")
      B = Number(str_B.slice(0, str_B.length - 1));
    else {
      result = -2;
      break;
    }
  }
  result += 1;
}

console.log(result + 1);
